From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Backport <backport@localhost>
Date: Wed, 21 Oct 2026 20:07:39 -0400
Subject: [PATCH 2/3] Define ZSTR_MAX_LEN in Zend/zend_string.h

Source: 0156-Define-ZSTR_MAX_LEN-Zend-zend_string.h.patch from Freexian ELTS php7.0 (prerequisite of the CVE-2026-109618/109619 backport).
---
 Zend/zend_string.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Zend/zend_string.h b/Zend/zend_string.h
index 784b88ea6a..5f7a9ea2b3 100644
--- a/Zend/zend_string.h
+++ b/Zend/zend_string.h
@@ -74,6 +74,9 @@ END_EXTERN_C()
 
 #define _ZSTR_STRUCT_SIZE(len) (_ZSTR_HEADER_SIZE + len + 1)
 
+#define ZSTR_MAX_OVERHEAD (ZEND_MM_ALIGNED_SIZE(_ZSTR_HEADER_SIZE + 1))
+#define ZSTR_MAX_LEN (SIZE_MAX - ZSTR_MAX_OVERHEAD)
+
 #define ZSTR_ALLOCA_ALLOC(str, _len, use_heap) do { \
 	(str) = (zend_string *)do_alloca(ZEND_MM_ALIGNED_SIZE_EX(_ZSTR_STRUCT_SIZE(_len), 8), (use_heap)); \
 	GC_REFCOUNT(str) = 1; \
